fvm/crypto/crypto.go

```diff
@@ -135,7 +139,7 @@ func VerifySignatureFromRuntime(
 	// check ECDSA compatibilites
 	if sigAlgo == crypto.ECDSAP256 || sigAlgo == crypto.ECDSASecp256k1 {
 		// hashing compatibility
-		if hashAlgo != hash.SHA2_256 && hashAlgo != hash.SHA3_256 {
+		if hashAlgo != hash.SHA2_256 && hashAlgo != hash.SHA3_256 && hashAlgo != hash.Keccak_256 {
```

Comment:
is this something acceptable for ECDSA signatures? I guessed there might be many ethereum signatures that can be verified by this

Reply:
We need to distinguish two different types of ECDSA signatures in the FVM:
 - the transaction signatures : for these, only SHA2-256 and SHA3-256 are supported for now. Keccak-256 can be added as well, though this will need changes in the account format in many places in our code/ecosystem. I think we can avoid that for now.
 - the contract signatures: signatures to be verified on chain by Cadence contracts. There is no problem for these signatures to support Keccak-256.  
 
The issue is that the FVM uses the same code to verify both these signatures above. I think it would be great to separate these 2 verifications as they have different requirements (a new issue to be created), and then we can support Keccak-256 for the second type. For now I suggest not to add Keccak-256

Reply:
I just update the code at d56bb0b to limit the transaction signatures hashing to specific ones, let me know if this addresses your concern, otherwise I can drop this for now.

Reply:
My previous comment confirms that this current file is fine. We can revert the commit https://github.com/onflow/flow-go/commit/d56bb0b58b73857e1d7eeac0066cf7d48b0759d1, my bad!